zerver/lib/integrations.py

```diff
@@ -404,6 +405,18 @@ def __init__(self, name: str, *args: Any, **kwargs: Any) -> None:
         logo="images/integrations/logos/github.svg",
         function="zerver.webhooks.github.view.api_github_webhook",
         stream_name="github",
+        config_options=[
+            (
+                "branches",
+                "Filter by branches (comma-separated branch names)",
+                check_string,
+            ),
+            (
+                "ignore_private_repositories",
+                "Exclude notifications from private repositories",
+                check_bool,
+            ),
+        ],
```

Comment:
Tuples are not very good types to use, make a `dataclass` instead.

Reply:
Also, probably we want the configuration options to be declared in the webhook file itself, so the set of parameters is contained and versioned along with the code there. So I'd define a dataclass in `zerver/lib/webhooks/common.py`, and then have the GitHub view export the values in some way like how event types are exported.